src/deploydocs.jl

```diff
@@ -379,14 +379,16 @@ function git_push(
         # Get the parts of the repo path and create upstream repo path
         user, host, upstream = user_host_upstream(repo)
 
-        keyfile = abspath(joinpath(root, ".documenter"))
+        keyfile = abspath(joinpath(homedir(), ".documenter-identity-file.tmp"))
```

Comment:
Better with temp?
```suggestion
        keyfile = abspath(joinpath(mktempdir(), ".documenter-identity-file.tmp"))
```

Reply:
Yeah, I think it's not inconceivable someone might try to upload the entire home directory, e.g. on a CI service where there's nothing much there anyways. If we wanted to be on the safe side, maybe even check if `tempdir` is a subdirectory of the target and error loudly if it is?

Reply:
`tempdir` would indeed be the most convenient, but what I was concerned about is that, if somehow it doesn't get cleaned up, it's better to leave the file into `/home` (where hopefully only the current user can read it), rather than `/tmp`. Especially if you're maybe running Documenter on 

Checking for the subdirectory seems reasonable though.

Reply:
Could the file not be created as read-write only for current user, then after the key is written, change to read-only for current user?

Reply:
I guess the permissions are fine already (user-only rw protects as much). I guess my main concern is some shared system where you might have the keyfile sitting around in the `/tmp` of some random node. In your home directory you might be more likely to notice it.